src/value/mod.rs

```diff
@@ -262,7 +262,7 @@ pub trait Element {
     type SymbolToken: SymbolToken + ?Sized;
     type Sequence: Sequence<Element = Self> + ?Sized;
     type Struct: Struct + ?Sized;
```

Comment:
*Minor*: Sine you made a similar change in another part of the code, we probably want `Struct<SymbolToken=Self::SymbolToken, Element=Self>`